rpcserver.go

```diff
@@ -5563,17 +5563,15 @@ func handleTxFeeInfo(s *rpcServer, cmd interface{}, closeChan <-chan struct{}) (
 // handleValidateAddress implements the validateaddress command.
 func handleValidateAddress(s *rpcServer, cmd interface{}, closeChan <-chan struct{}) (interface{}, error) {
 	c := cmd.(*dcrjson.ValidateAddressCmd)
-
 	result := dcrjson.ValidateAddressChainResult{}
 	addr, err := dcrutil.DecodeAddress(c.Address)
 	if err != nil {
 		// Return the default value (false) for IsValid.
-		return result, nil
+		return result, err
 	}
 
 	result.Address = addr.EncodeAddress()
-	result.IsValid = true
-
+	result.IsValid = addr.IsForNet(s.server.chainParams)
```

Comment:
This should return an error when it's not for the current network as opposed to just being false.  It is still a valid address, just not for the current network, so just returning false in that case would be misleading and incorrect.

It should match other handlers which do:

```Go
	if !addr.IsForNet(s.server.chainParams) {
		return nil, rpcAddressKeyError("Wrong network: %v", addr)
	}
}
```